test/e2e/storage/dax/dax.go

```diff
@@ -0,0 +1,247 @@
+
+	. "github.com/onsi/ginkgo"
+)
+
+type daxTestSuite struct {
+	tsInfo testsuites.TestSuiteInfo
+}
+
+var _ testsuites.TestSuite = &daxTestSuite{}
+
+// InitDaxTestSuite returns daxTestSuite that implements TestSuite interface
+func InitDaxTestSuite() testsuites.TestSuite {
+	return &daxTestSuite{
+		tsInfo: testsuites.TestSuiteInfo{
+			Name: "dax",
+			TestPatterns: []testpatterns.TestPattern{
+				testpatterns.DefaultFsDynamicPV,
+				testpatterns.Ext4DynamicPV,
+				testpatterns.XfsDynamicPV,
+
+				testpatterns.DefaultFsEphemeralVolume,
+				testpatterns.Ext4EphemeralVolume,
+				testpatterns.XfsEphemeralVolume,
+
+				testpatterns.BlockVolModeDynamicPV,
+			},
+		},
+	}
+}
+
+func (p *daxTestSuite) GetTestSuiteInfo() testsuites.TestSuiteInfo {
+	return p.tsInfo
+}
+
+func (p *daxTestSuite) SkipRedundantSuite(driver testsuites.TestDriver, pattern testpatterns.TestPattern) {
+}
+
+type local struct {
+	config      *testsuites.PerTestConfig
+	testCleanup func()
+
+	resource       *testsuites.VolumeResource
+	root           string
+	daxCheckBinary string
+}
+
+func (p *daxTestSuite) DefineTests(driver testsuites.TestDriver, pattern testpatterns.TestPattern) {
+	var l local
+
+	f := framework.NewDefaultFramework("dax")
+
+	init := func() {
+		l = local{}
+
+		// Build pmem-dax-check helper binary.
+		l.root = os.Getenv("REPO_ROOT")
+		l.daxCheckBinary = "_work/pmem-dax-check"
+		build := exec.Command("/bin/sh", "-c", os.Getenv("GO")+" build -o "+l.daxCheckBinary+" ./test/cmd/pmem-dax-check")
+		build.Stdout = GinkgoWriter
+		build.Stderr = GinkgoWriter
+		build.Dir = l.root
+		By("Compiling with: " + strings.Join(build.Args, " "))
+		err := build.Run()
+		framework.ExpectNoError(err, "compile ./test/cmd/pmem-dax-check")
+
+		// Now do the more expensive test initialization.
+		l.config, l.testCleanup = driver.PrepareTest(f)
+		l.resource = testsuites.CreateVolumeResource(driver, l.config, pattern)
+	}
+
+	cleanup := func() {
+		if l.resource != nil {
+			l.resource.CleanupResource()
+			l.resource = nil
+		}
+
+		if l.testCleanup != nil {
+			l.testCleanup()
+			l.testCleanup = nil
+		}
+	}
+
+	It("should support MAP_SYNC", func() {
+		init()
+		defer cleanup()
+
+		l.testDaxInPod(f, l.resource.Pattern.VolMode, l.resource.VolSource, l.config)
+	})
+}
+
+func (l local) testDaxInPod(
+	f *framework.Framework,
+	volumeMode v1.PersistentVolumeMode,
+	source *v1.VolumeSource,
+	config *testsuites.PerTestConfig) {
+
+	const (
+		volPath       = "/vol1"
+		volName       = "vol1"
+		containerName = "dax-container"
+	)
+	privileged := volumeMode == v1.PersistentVolumeBlock
+	pod := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "dax-volume-test",
+			Namespace: f.Namespace.Name,
+		},
+		Spec: v1.PodSpec{
+			Containers: []v1.Container{
+				{
+					SecurityContext: &v1.SecurityContext{
+						Privileged: &privileged,
+					},
+					Name:    containerName,
+					Image:   os.Getenv("PMEM_CSI_IMAGE"),
```

Comment:
I think we may use any generic(maybe ubuntu) image here which has `mknod` and other filesystem utilities installed right. `PMEM_CSI_IMAGE` here is bit misleading. Running this dax test outside our driver image is preferable in my opinion.

Reply:
Then we need to pull another, potentially large image into the cluster. We also don't have control over the content of that image and would have to lock onto a specific, soon out-dated version to avoid a potential regression should "ubuntu:latest" ever drop required tools (like mkfs.ext4).

I chose the PMEM-CSI image because it's already on the cluster and we know that it has the necessary tools.

Reply:
Ok, let's keep this way.